docs/the-new-architecture/pillars-codegen.md

```diff
@@ -3,5 +3,227 @@ id: pillars-codegen
+                            ├── ComponentDescriptors.h
+                            ├── EventEmitters.cpp
+                            ├── EventEmitters.h
+                            ├── Props.cpp
+                            ├── Props.h
+                            ├── RCTComponentViewHelpers.h
+                            ├── ShadowNodes.cpp
+                            └── ShadowNodes.h
+```
+
+At the root of the hierarchy, we have the `codegen` folder, as expected. Nested into it we have a couple more folders: `build/generated`.
+
+Then, we have an `ios` folder which contains:
+
+- A custom folder for each TurboModule.
+- The header (`.h`) and implementation (`.mm`) files for the `RCTThirdPartyFabricComponentsProvider`.
+- A base `react/renderer/components` folder which contains a custom folder for each `Fabric Component`.
+
+In this hierarchy, there are a TurboModule and a set of Fabric Components that are generated by React Native itself: `FBReactNativeSpec` and `rncore`. These will appear even if we don't have any extra TurboModule or Fabric Component: React Native requires them in order to work properly.
+
+### TurboModules
+
+The various TurboModules folders contain two files for each TurboModule: an interface file and an implementation file.
+
+The interface files have the same name of the TurboModule and they contain methods to initialize their the JSI interface.
+
+The implementation files, instead, have the `-generated` suffix and they contains the logic to invoke the native methods from JS and viceversa.
+
+### Fabric Components
+
+The content of each Fabric Component folder contains several files. The basic element for a Fabric Componenent is the `ShadowNode`: it represents a node in the React absract tree. The `ShadowNode` represents a React entity, therefore it could need some props, which are defined in the `Props` files and, sometimes, an `EventEmitter`, defined in the corresponding file.
+
+Additionally, the **CodeGen** also creates a `ComponentDescriptor.h` and an `RCTComponentViewHelpers.h` files: the first one is used by React Native and Fabric to properly get a reference to the Component, while the latter contains some helper methods and protocols that can be implemented by the Native View to properly respond to JSI invocations.
+
+For further details about how Fabric works, have a look at the [Renderer](../../architecture/fabric-renderer) section.
+
+### RCTThirdPartyFabricComponentsProvider
+
+These are an interface and an implementation files for a registry. React Native access this registry at runtime to retrieve the right class for a required Fabric Component. Once React Native has an handle to that class, it can instantiate it.
+
+# Android
+
+## Running the CodeGen
+
+Android `CodeGen` relies on a gradle task to generate the required code. First, we need to configure the Android app to work with the New Architecture, otherwise the gradle task will fail. We can configure the app by:
```

Comment:
```suggestion
Android `CodeGen` relies on a Gradle task to generate the required code. First, we need to configure the Android app to work with the New Architecture, otherwise the Gradle task will fail. We can configure the app by:
```